src/components/base/JTextArea/JTextArea.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+import React, { PureComponent } from 'react'
+import { omit } from 'lodash-es'
+import classNames from 'classnames'
+
+import jTextAreaStyle from './jTextArea.m.scss'
+
+type Theme = 'blue'
+
+type Props = StyleComponent<Theme> & {
+  ...HTMLTextAreaElement,
+  onChange: Function,
+  onBlur?: Function,
+  onFocus?: Function,
+}
+
+const MAX_ROWS = 12
+
+function heightCalc(event: SyntheticEvent<HTMLTextAreaElement>): void {
+  if (event.currentTarget) {
+    const target = event.currentTarget
+
+    while (target.clientHeight < target.scrollHeight && target.rows <= MAX_ROWS) {
+      // mutating the DOM
+      // eslint-disable-next-line fp/no-mutation
+      target.rows += 1
+    }
+  }
+}
+
+function handleChange(finalFormHandler: Function): SyntheticEvent<HTMLTextAreaElement> => void {
+  return function textAreaChangeHandler(fieldEvent) {
+    heightCalc(fieldEvent)
+    finalFormHandler(fieldEvent)
+  }
+}
+
+export class JTextArea extends PureComponent<Props, void> {
```

Comment:
This component is too specific, to be named `JTextArea` and to come under `base` namespace. Consider moving it and renaming according to its usage context.